src/neo/Persistence/Memory/Snapshot.cs

```diff
@@ -0,0 +1,69 @@
+using Neo.IO;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+
+namespace Neo.Persistence.Memory
+{
+    internal class Snapshot : ISnapshot
+    {
+        private readonly ReaderWriterLockSlim readerWriterLock;
+        private readonly Dictionary<byte[], byte[]>[] innerData;
+        private readonly ImmutableDictionary<byte[], byte[]>[] immutableData;
+        private readonly Dictionary<byte[], byte[]>[] writeBatch;
+
+        public Snapshot(Dictionary<byte[], byte[]>[] innerData, ReaderWriterLockSlim readerWriterLock)
+        {
+            this.readerWriterLock = readerWriterLock;
+            this.innerData = innerData;
+            readerWriterLock.EnterReadLock();
+            this.immutableData = innerData.Select(p => p.ToImmutableDictionary(ByteArrayEqualityComparer.Default)).ToArray();
+            readerWriterLock.ExitReadLock();
+            this.writeBatch = new Dictionary<byte[], byte[]>[innerData.Length];
```

Comment:
We can change all to `ConcurrentDictionary` and remove the `readerWriterLock`

Reply:
Good idea! I will update it.

Reply:
Done.